core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -939,4 +943,87 @@ public Object transformResult(Object[] results) {
       }
    }
 
+   @Override
+   public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredCommand command) throws Throwable {
+      // Lifespan expiration just behaves like a remove command
+      if (!command.isMaxIdle()) {
+         return visitRemoveCommand(ctx, command);
+      }
+
+      Object key = command.getKey();
+      CacheEntry entry = ctx.lookupEntry(key);
+
+      if (isLocalModeForced(command)) {
+         if (entry == null) {
+            entryFactory.wrapExternalEntry(ctx, key, null, false, true);
```

Comment:
This is essentially a copy from `BaseDistributionInterceptor#handleNonTxWriteCommand` for this block.

Reply:
IMO it's not necessary, if the command has the CACHE_MODE_LOCAL flag it's EWI's job to wrap a null entry in the context. Unfortunately, @rvansa didn't say anything about CACHE_MODE_LOCAL in the EntryFactory javadoc with the wrapping patterns.

Reply:
I don't think it meant "clustered events in this context" I think it was just event in general, like messages or such. This message has been around forever :)

I will remove the null check though.

Reply:
Right, I thought it might have had something to do with clustered events at some point and the method got extracted+reused. It makes even less sense to me now that I know it's been that way forever :)